src/NuGet.Services.AzureSearch/SearchService/SearchResponseBuilder.cs

```diff
@@ -443,8 +443,7 @@ private string GetIconUrl(IBaseMetadataDocument document)
 
         private V3SearchDeprecation GetV3SearchDeprecation(SearchDocument.Full document)
         {
-            if (document.Deprecation != null && document.Deprecation.Reasons != null && document.Deprecation.Reasons.Length > 0)
-            //according to nuget api, reasons array is required and cannot be empty
+            if (HasValidDeprecation(document.Deprecation))
             {   
                 var deprecation = new V3SearchDeprecation();
                 deprecation.AlternatePackage = new V3SearchAlternatePackage{}; 
```

Comment:
I think this should actually look like this (as well as for V2). Correct me if I'm wrong!
```suggestion
                deprecation.AlternatePackage = new V3SearchAlternatePackage(); 
```

Reply:
Yes, make sense, we are not assigning anything there.